doc/message-registry.md

```diff
@@ -41,7 +41,7 @@ There are currently 54 entries.
 | [XKB-378] | `duplicate-entry` | An entry is duplicated and will be ignored | Warning |
 | [XKB-386] | `recursive-include` | Included files form cycle | Error |
 | [XKB-407] | `conflicting-key-type-definitions` | Conflicting definitions of a key type | Warning |
-| [XKB-428] | `wrong-scope` | A statement is in a wrong scope and should be moved | Error |
+| [XKB-428] | `global-defaults-wrong-scope` | A global defaults statement is in a wrong scope and should be moved | Error |
```

Comment:
I have a more general question: some mistakes in XKB files are treated as allowable and cause the corresponding statement to be skipped, while others fail the parsing.

What is the guiding rule for assigning mistakes into these two categories? I really hope it's not "whatever current parser can parse is not an error", as it ensrines the (undocumented) parser implementation designs as a specification.

Reply:
> I really hope it's not "whatever current parser can parse is not an error", as it ensrines the (undocumented) parser implementation designs as a specification.

I fear this is the case, because there is no formal documentation for the XKB _text format_. E.g. in this PR, the keymaps we test in `data/test` are parsed correctly with `xkbcomp`[^1] but will fail with `xkbcommon`.

In this specific case I see no reason to _not_ reject the file: this error encompasses both a wrong scope and invalid qualifiers (e.g. `garbage.`).

In my opinion XKB is enough complex so we should not be too lenient.

[^1]: Well, it parses if you fix `xkb_compat    { include "comple" };`.